sycl/source/handler.cpp

```diff
@@ -27,8 +28,124 @@ namespace sycl {
 
 handler::handler(std::shared_ptr<detail::queue_impl> Queue, bool IsHost)
     : MQueue(std::move(Queue)), MIsHost(IsHost) {
-  MSharedPtrStorage.emplace_back(
-      std::make_shared<std::vector<detail::ExtendedMemberT>>());
+  // Create extended members and insert handler_impl
+  // TODO: When allowed to break ABI the handler_impl should be made a member
+  //       of the handler class.
+  auto ExtendedMembers =
+      std::make_shared<std::vector<detail::ExtendedMemberT>>();
+  detail::ExtendedMemberT HandlerImplMember = {
+      detail::ExtendedMembersType::HANDLER_IMPL,
+      std::make_shared<detail::handler_impl>()};
+  ExtendedMembers->push_back(std::move(HandlerImplMember));
+  MSharedPtrStorage.push_back(std::move(ExtendedMembers));
+}
+
+/// Gets the handler_impl at the start of the extended members.
+// detail::GlobalHandler::instance().getHandlerExtendedMembersMutex() must
+// be held when calling this function.
+std::shared_ptr<detail::handler_impl>
+getHandlerImpl(const std::shared_ptr<std::vector<detail::ExtendedMemberT>>
+                   &ExtendedMembersVec) {
+  assert(ExtendedMembersVec->size() > 0);
+
+  auto HandlerImplMember = (*ExtendedMembersVec)[0];
+
+  assert(detail::ExtendedMembersType::HANDLER_IMPL == HandlerImplMember.MType);
+
+  return std::static_pointer_cast<detail::handler_impl>(
+      HandlerImplMember.MData);
+}
+
+// Common implementation for getting/inserting handler kernel bundle.
+// detail::GlobalHandler::instance().getHandlerExtendedMembersMutex() must
+// be held when calling this function.
+std::shared_ptr<detail::kernel_bundle_impl>
+getOrInsertHandlerKernelBundleCommon(
```

Comment:
Are there any calls to this function?

Reply:
Sorry, I forgot to remove it. It is no more!